### ISSUE TITLE
Ads on Facebook

### ISSUE THREAD

**Issue by ragibkl** [ragibkl/adblock-dns-server#13]
A user has reported that ads on Facebook are not blocked by this dns. Should take a look.

**Comment by TechComputerServiceDunkerque** [comment]
I was told to create a host with this list to block facebook ads, but I don't know how to do it.  

https://www.fanboy.co.nz/fanboy-antifacebook.txt

**Comment by ragibkl** [comment]
@TechComputerServiceDunkerque 

I took a look at the antifacebook file that you shared. It seems that the format is not a hosts format. It looks like adblock plus extension config file or something similar. I don't think we can use this file.

Do you have any example of what kind of ads do you see on facebook? Maybe you can share some screenshot to me, and I can try to debug what domains are being called. If it is some isolated domains, we can block them.

**Comment by TechComputerServiceDunkerque** [comment]
https://image.noelshack.com/fichiers/2020/08/6/1582410676-s00222-23295244.jpg
https://image.noelshack.com/fichiers/2020/08/6/1582410687-s00222-23300573.jpg

**Comment by Tomatoide** [comment]
I have a question, in the regex whitelist, xx.fbcdn.com is whitelisted, also there are 2 other whitelists for facebook and fbcdn, but if you check a blocklist eg stevenblack, you can find `sonar-iad.xx.fbcdn.net`, but because the whitelists it's not blocked, so is that maybe a reason for this issue?

**Comment by ragibkl** [comment]
@Tomatoide ,

This issue is way longer than when I implemented the regex whitelist. I think fbcdn.net should be okay to whitelist. The name says CDN so it's probably for loading static resources such as image assets, logos, static javascript, or maybe themes and fonts.

The thing is, I haven't used facebook much, and I haven't done any debugging to figure out, if there is a specific domain we can use to block this types of ads.

We need to dig deeper if we want to identity the correct domain to block.

**Comment by ragibkl** [comment]
I did a basic lookup just now. This is for a post in FB with a `Sponsored` label, which I believe is what is reported in this issue.
```
POST https://www.facebook.com/api/graphql/
POST https://www.facebook.com/ajax/bulk-route-definitions/
```

The ads are loaded using `www.facebook.com` domain. So it means that we can't really block this without also effectively blocking facebook access.

tldr; I don't know how to block these ads..

**Comment by Tomatoide** [comment]
From what I read apps nowadays are loading ads from the same servers that load content, so yes it can say cdn but would also load ads, so its hard to block all ads especially with dns based adblock, but you can block some of them if you can figure out some specific subdomains like the one I mentioned before in blocklists, I don't use facebook that much either but thought maybe whitelisting everything is counterproductive because you have lists that are neutralized that can fix the issue even if partially.

**Comment by ragibkl** [comment]
I agree with conclusion. We cannot use DNS to block this, since its from the same servers as Facebook. Let's close this and move one.
